plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,623 @@
+        type: str
+        choices: [local, nfs, cifs, http, https]
+        default: local
+      file_name:
+        description:
+          - Diagnostics file name for I(export) operation.
+        type: str
+      ip_address:
+        description:
+          - IP address of the network share.
+          - I(ip_address) is required when I(share_type) is C(nfs), C(cifs), C(http) or C(https).
+        type: str
+      share_name:
+        description:
+          - Network share or local path of the diagnostics file.
+        type: str
+      workgroup:
+        description:
+          - Workgroup of the network share.
+          - I(workgroup) is applicable only when I(share_type) is C(cifs).
+        type: str
+      username:
+        description:
+          - Username of the network share.
+          - I(username) is required when I(share_type) is C(cifs).
+        type: str
+      password:
+        description:
+          - Password of the network share.
+          - I(password) is required when I(share_type) is C(cifs).
+        type: str
+      ignore_certificate_warning:
+        description:
+          - Ignores the certificate warning while connecting to Share and is only applicable when I(share_type) is C(https).
+          - C(off) ignores the certificate warning.
+          - C(on) does not ignore the certificate warning.
+        type: str
+        choices: ["off", "on"]
+        default: "off"
+      proxy_support:
+        description:
+          - Specifies if proxy support must be used or not.
+          - C(off) does not use proxy settings.
+          - C(default_proxy) uses the default proxy settings.
+          - C(parameters_proxy) uses the specified proxy settings. I(proxy_server) is required when I(proxy_support) is C(parameters_proxy).
+          - I(proxy_support) is only applicable when I(share_type) is C(https) or C(https).
+        type: str
+        choices: ["off", "default_proxy", "parameters_proxy"]
+        default: "off"
+      proxy_type:
+        description:
+          - The proxy type of the proxy server.
+          - C(http) to select HTTP proxy.
+          - C(socks) to select SOCKS proxy.
+          - I(proxy_type) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
+        choices: [http, socks]
+        default: http
+      proxy_server:
+        description:
+          - The IP address of the proxy server.
+          - I(proxy_server) is required when I(proxy_support) is C(parameters_proxy).
+          - I(proxy_server) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
+      proxy_port:
+        description:
+          - The port of the proxy server.
+          - I(proxy_port) is only applicable when I(share_type) is C(https) or C(https) and when I(proxy_support) is C(parameters_proxy).
+        type: str
```

Comment:
If payload is accepting this parameter as string, then make required handling in the code

Reply:
Done